### PR TITLE
Fix onboarding integration tests

### PR DESCRIPTION
**Pull request by Bilb**

### CODE REVIEW

File: ts/mains/main_node.ts [hunk 1]
```diff
@@ -995,6 +995,28 @@ ipc.on('password-window-login', async (event, passPhrase) => {
   }
 });
 
+ipc.on('password-recovery-phrase', async (event, passPhrase) => {
```
Comment (yougotwill) [inline ts/mains/main_node.ts:998]: Did this exist before onboarding and I somehow removed it by mistake? It looks familar 🤔

File: ts/components/settings/section/CategoryRecoveryPassword.tsx [hunk 2]
```diff
@@ -68,9 +67,11 @@ const qrLogoProps: QRCodeLogoProps = {
 };
 
 export const SettingsCategoryRecoveryPassword = () => {
-  const [loadingSeed, setLoadingSeed] = useState(true);
-  const [recoveryPhrase, setRecoveryPhrase] = useState('');
-  const [hexEncodedSeed, setHexEncodedSeed] = useState('');
+  const recoveryPhrase = getCurrentRecoveryPhrase();
+  if (!recoveryPhrase || isEmpty(recoveryPhrase)) {
```
Comment (yougotwill) [inline ts/components/settings/section/CategoryRecoveryPassword.tsx:71]: I guess we don't need isEmpty anymore right?

File: preload.js [hunk 3]
```diff
@@ -72,6 +71,18 @@ window.setPassword = async (passPhrase, oldPhrase) =>
     ipc.send('set-password', passPhrase, oldPhrase);
   });
 
+// called to verify that the password is correct when showing the recovery from seed modal
+window.onTryPassword = passPhrase =>
+  new Promise((resolve, reject) => {
```
Comment (yougotwill) [inline preload.js:76]:
```suggestion
window.onTryPassword = async passPhrase =>
```

**Review by yougotwill — CHANGES_REQUESTED** [review]
When I remove a password it does not close the modal or show the success toast. If you close the modal manually it stills for a password everywhere. However if you restart the app then the password is no longer needed. So Basically it is removing it silently but also not refreshing the renderer information onPasswordVerified I guess

File: ts/components/SessionPasswordPrompt.tsx [hunk 4]
```diff
@@ -112,10 +112,9 @@ class SessionPasswordPromptInner extends PureComponent<unknown, State> {
   }
 
   public async onLogin(passPhrase: string) {
-    const passPhraseTrimmed = passPhrase.trim();
-
+    // Note: we don't trim the password anymore. If the user entered a space at the end, so be it.
```
[the remark this answers is not in the document]
Comment (Bilb) [reply ts/components/SessionPasswordPrompt.tsx:115]: we don't want trimming at all, so why add it to the input?

File: ts/test/session/unit/utils/job_runner/JobRunner_test.ts [hunk 5]
```diff
@@ -303,6 +304,8 @@ describe('JobRunner', () => {
 
       await job2.waitForCurrentTry();
       await runnerMulti.waitCurrentJob();
+      // we need to give sometime to the jobrunner to handle the return of job2 and remove it
+      await sleepFor(100);
```
[the remark this answers is not in the document]
Comment (Bilb) [reply ts/test/session/unit/utils/job_runner/JobRunner_test.ts:308]: I'll assume you didn't want to swap those 2 lines in the suggestion

File: ts/util/passwordUtils.ts [hunk 6]
```diff
@@ -14,27 +14,26 @@ const sha512 = (text: string) => {
...
   }
 
-  const trimmed = phrase.trim();
-  if (trimmed.length === 0) {
+  if (phrase.length === 0) {
     return window?.i18n ? window?.i18n('noGivenPassword') : ERRORS.LENGTH;
   }
 
-  if (trimmed.length < 6 || trimmed.length > MAX_PASSWORD_LENGTH) {
+  if (phrase.length < 6 || phrase.length > MAX_PASSWORD_LENGTH) {
```
[the remark this answers is not in the document]
Comment (Bilb) [reply ts/util/passwordUtils.ts:30]: added a TODO

File: ts/test/session/unit/onion/SnodePoolUpdate_test.ts [hunk 7]
```diff
@@ -67,10 +67,10 @@ describe('OnionPaths', () => {
       expect(fetched).to.deep.equal(fakeSnodePool);
     });
 
-    it('if the cached snode pool 12 or less snodes, trigger a fetch from the seed nodes', async () => {
-      const length12 = fakeSnodePool.slice(0, 12);
-      expect(length12.length).to.eq(12);
-      getSnodePoolFromDb = stubData('getSnodePoolFromDb').resolves(length12);
+    it('if the cached snode pool 8 or less snodes, trigger a fetch from the seed nodes', async () => {
+      const length8 = fakeSnodePool.slice(0, 8);
```
[the remark this answers is not in the document]
Comment (Bilb) [reply ts/test/session/unit/onion/SnodePoolUpdate_test.ts:71]: changed, but I assumed you didn't mean to swap the 2 lines

File: ts/test/session/unit/onion/GuardNodes_test.ts [hunk 8]
```diff
@@ -51,7 +52,7 @@ describe('GuardNodes', () => {
       Sinon.restore();
     });
 
-    it('does not fetch from seed if we got 12 or more snodes in the db', async () => {
+    it('does not fetch from seed if we got 8 or more snodes in the db', async () => {
       stubData('getSnodePoolFromDb').resolves(fakeSnodePool);
```
[the remark this answers is not in the document]
Comment (Bilb) [reply ts/test/session/unit/onion/GuardNodes_test.ts:56]: fixed

File: ts/test/session/unit/onion/SnodePoolUpdate_test.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (Bilb) [reply ts/test/session/unit/onion/SnodePoolUpdate_test.ts:71]: I think that's a gh issue

File: ts/test/session/unit/utils/job_runner/JobRunner_test.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (Bilb) [reply ts/test/session/unit/utils/job_runner/JobRunner_test.ts:308]: just a gh issue probably

File: ts/test/session/unit/onion/GuardNodes_test.ts [hunk 9]
```diff
@@ -168,13 +166,14 @@ describe('GuardNodes', () => {
       // run the command
       const guardNodes = await OnionPaths.selectGuardNodes();
 
-      expect(guardNodes.length).to.be.equal(3);
-      expect(testGuardNode.callCount).to.be.equal(3);
+      expect(guardNodes.length).to.be.equal(2);
+      expect(testGuardNode.callCount).to.be.equal(2);
```
[the remark this answers is not in the document]
Comment (Bilb) [reply ts/test/session/unit/onion/GuardNodes_test.ts:170]: added a comment

File: ts/session/utils/job_runners/jobs/ConfigurationSyncJob.ts [hunk 10]
```diff
@@ -23,8 +23,8 @@ import {
 } from '../PersistedJob';
 import { DURATION } from '../../../constants';
 
-const defaultMsBetweenRetries = 15000; // a long time between retries, to avoid running multiple jobs at the same time, when one was postponed at the same time as one already planned (5s)
-const defaultMaxAttempts = 2;
+const defaultMsBetweenRetries = 5000; // a long time between retries, to avoid running multiple jobs at the same time, when one was postponed at the same time as one already planned (5s)
```
[the remark this answers is not in the document]
Comment (Bilb) [reply ts/session/utils/job_runners/jobs/ConfigurationSyncJob.ts:26]: updated

File: ts/session/utils/job_runners/jobs/ConfigurationSyncJob.ts [hunk 11]
```diff
@@ -332,18 +318,18 @@ async function queueNewJobIfNeeded() {
     !lastRunConfigSyncJobTimestamp ||
     lastRunConfigSyncJobTimestamp < Date.now() - defaultMsBetweenRetries
   ) {
-    // window.log.debug('Scheduling ConfSyncJob: ASAP');
-    // we postpone by 1000ms to make sure whoever is adding this job is done with what is needs to do first
+    // Note: we postpone by 3s for two reasons:
+    // - to make sure whoever is adding this job is done with what is needs to do first
+    // - to allow a just created device to process incoming config messages before pushing a new one
     // this call will make sure that there is only one configuration sync job at all times
```
[the remark this answers is not in the document]
Comment (Bilb) [reply ts/session/utils/job_runners/jobs/ConfigurationSyncJob.ts:324]: Assumed you meant `a new one` at the end

File: ts/session/utils/job_runners/PersistedJob.ts [hunk 12]
```diff
@@ -108,7 +108,19 @@ export abstract class PersistedJob<T extends PersistedJobData> {
...
+      // eslint-disable-next-line more/no-then
+      this.runningPromise = this.run()
+        .then(jobResult => {
+          this.runningPromise = null;
+          return jobResult;
+        })
+        .catch(e => {
+          window.log.warn(
+            'runJob() threw. this cannot happen, but rehtrowing as this should be handled in each jobs run()',
+            e
```
[the remark this answers is not in the document]
Comment (Bilb) [reply ts/session/utils/job_runners/PersistedJob.ts:120]: this doesn't mean anything to me. I like mine better

File: ts/mains/main_node.ts (hunk 1 above)
Comment (Bilb) [reply ts/mains/main_node.ts:998]: no it didn't. It's brand new

File: ts/components/settings/section/CategoryRecoveryPassword.tsx (hunk 2 above)
Comment (Bilb) [reply ts/components/settings/section/CategoryRecoveryPassword.tsx:71]: yep good catch. It's a string so falsy when empty

File: preload.js (hunk 3 above)
Comment (Bilb) [reply preload.js:76]: fixed

File: preload.js [hunk 13]
```diff
@@ -72,6 +71,18 @@ window.setPassword = async (passPhrase, oldPhrase) =>
...
+    ipcRenderer.once('password-recovery-phrase-response', (event, error) => {
+      if (error) {
```
[the remark this answers is not in the document]
Comment (Bilb) [reply preload.js:78]: renamed to `_event` as we have lodash imported as _

File: ts/components/SessionPasswordPrompt.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (yougotwill) [reply ts/components/SessionPasswordPrompt.tsx:115]: Ah I see some people would want whitespace as part of their password. All good resolving

File: ts/session/utils/job_runners/PersistedJob.ts (hunk 12 above)
[the remark this answers is not in the document]
Comment (yougotwill) [reply ts/session/utils/job_runners/PersistedJob.ts:120]: Cool but you have mispelt "rehtrowing"

File: ts/session/utils/job_runners/jobs/ConfigurationSyncJob.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (yougotwill) [reply ts/session/utils/job_runners/jobs/ConfigurationSyncJob.ts:324]: Yeah vs code did something weird

File: ts/test/session/unit/onion/SnodePoolUpdate_test.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (yougotwill) [reply ts/test/session/unit/onion/SnodePoolUpdate_test.ts:71]: Yeah probably vs code. I'm doing my reviews using the extension. My b

File: ts/test/session/unit/utils/job_runner/JobRunner_test.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (yougotwill) [reply ts/test/session/unit/utils/job_runner/JobRunner_test.ts:308]: Damn sorry man. Yeah did not want to swap them

**Review by yougotwill — CHANGES_REQUESTED** [review]
Commented on the one spelling error

**Review by yougotwill — APPROVED** [review]
Nice job! LGTM 🚢

**Pull request merged**